service/test/endToEnd/authorization/usersTest.js

```diff
@@ -0,0 +1,1929 @@
+const expect = require('code').expect
+const Lab = require('lab')
+const lab = exports.lab = Lab.script()
+const utils = require('../../utils')
+const userOps = require('../../../lib/ops/userOps')
+const policyOps = require('../../../lib/ops/policyOps')
+const teamOps = require('../../../lib/ops/teamOps')
+const server = require('./../../../wiring-hapi')
+
+
+lab.experiment('Routes Authorizations', () => {
+  lab.experiment('users', () => {
+    lab.experiment('GET /users/:id', () => {
+
+      const organizationId = 'WONKA'
+      let callerId
+      let calledId
+      let calledTeamId
+      let policyId
+
+      function Policy (Statement) {
+        return {
+          id: policyId || null,
+          version: '2016-07-01',
+          name: 'Test Policy',
+          statements: JSON.stringify({
+            Statement: Statement || [{
+              Effect: 'Allow',
+              Action: ['dummy'],
+              Resource: ['dummy']
+            }]
+          }),
+          organizationId
+        }
+      }
+
+      lab.before((done) => {
+        userOps.createUser({ name: 'caller', organizationId }, (err, caller) => {
+          if (err) return done(err)
+          callerId = caller.id
+
+          userOps.createUser({ name: 'called', organizationId }, (err, called) => {
+            if (err) return done(err)
+            calledId = called.id
+
+            teamOps.createTeam({ name: 'called team', organizationId }, (err, team) => {
+              if (err) return done(err)
+              calledTeamId = team.id
+
+              teamOps.addUsersToTeam({ id: calledTeamId, users: [calledId], organizationId }, (err) => {
+                if (err) return done(err)
+
+                const policyCreateData = Policy()
+
+                policyOps.createPolicy(policyCreateData, (err, policy) => {
+                  if (err) return done(err)
+                  policyId = policy.id
+
+                  userOps.addUserPolicies({ id: callerId, organizationId, policies: [policyId] }, done)
+                })
+              })
+            })
+          })
+        })
+      })
+
+      lab.after((done) => {
+        userOps.deleteUser({ id: callerId, organizationId }, (err) => {
```

Comment:
I assume after we refactor tests we'll use a tool to clean all db removing the manual delete. Like: [node-database-cleaner](https://github.com/emerleite/node-database-cleaner)

One issue that I see here is the cleanup running at the end. It could generate junk data. Cleaning should be done before.